src/grain/createGrainAllocation.test.js

```diff
@@ -0,0 +1,356 @@
+// @flow
+
+import {NodeAddress} from "../core/graph";
+import {ONE, ZERO, fromApproximateFloat, format} from "./grain";
+import {
+  GRAIN_ALLOCATION_VERSION_1,
+  createGrainAllocation,
+} from "./createGrainAllocation";
+import deepFreeze from "deep-freeze";
+import type {
+  CredHistory,
+  BalancedV1,
+  ImmediateV1,
+  GrainReceipt,
+  GrainAllocationV1,
+  AllocationStrategy,
+} from "./createGrainAllocation";
+import type {Grain} from "./grain";
+
+describe("src/grain/createGrainAllocation", () => {
+  const foo = NodeAddress.fromParts(["foo"]);
+  const bar = NodeAddress.fromParts(["bar"]);
+
+  const unevenInterval = deepFreeze({
+    intervalEndMs: 10,
+    cred: new Map([
+      [foo, 9],
+      [bar, 1],
+    ]),
+  });
+
+  const evenInterval = deepFreeze({
+    intervalEndMs: 20,
+    cred: new Map([
+      [foo, 1],
+      [bar, 1],
+    ]),
+  });
+
+  const singlePersonInterval = deepFreeze({
+    intervalEndMs: 30,
+    cred: new Map([[bar, 2]]),
+  });
+
+  const credHistory: CredHistory = deepFreeze([
+    unevenInterval,
+    evenInterval,
+    singlePersonInterval,
+  ]);
+
+  // there's no JSON serialization for BigInts, so we need to convert
+  // the BigInts before passing them into expect comparisons. Otherwise,
+  // when the tests fail, they'll fail with an unhelpful JSON serialization
+  // issue.
+  function fmt(g: Grain) {
+    return format(g, 3);
+  }
+  function safeReceipts(receipts: $ReadOnlyArray<GrainReceipt>) {
+    return receipts.map(({address, amount}) => ({
+      address,
+      amount: fmt(amount),
+    }));
+  }
```

Comment:
Nit: may want to sort the array by address as well to prevent test flakes.

Reply:
I think changing the order of the provided array would be unintended behaviour and might cause more test flakes. I would expect this function to return the results in the same order as I gave them. If we encounter flakiness we can update the function to sort, but until then it might just cause confusion.